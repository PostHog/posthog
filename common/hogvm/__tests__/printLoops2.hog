// Printing recursive objects.
let root := {'key': 'value', 'key2': 'value2'}
let leaf := {'key': 'value', 'key2': 'value2'}
for (let i := 0; i < 30; i := i + 1) {
  root[f'key_{i}'] := {
    'something': leaf,
  }
}
// Should NOT replace all leaves with nulls.
print(root)
print(jsonParse(jsonStringify(root)))
