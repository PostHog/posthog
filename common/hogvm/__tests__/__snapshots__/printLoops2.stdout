{'key': 'value', 'key2': 'value2', 'key_0': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_1': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_2': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_3': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_4': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_5': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_6': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_7': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_8': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_9': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_10': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_11': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_12': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_13': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_14': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_15': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_16': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_17': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_18': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_19': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_20': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_21': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_22': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_23': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_24': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_25': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_26': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_27': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_28': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_29': {'something': {'key': 'value', 'key2': 'value2'}}}
{'key': 'value', 'key2': 'value2', 'key_0': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_1': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_2': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_3': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_4': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_5': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_6': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_7': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_8': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_9': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_10': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_11': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_12': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_13': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_14': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_15': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_16': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_17': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_18': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_19': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_20': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_21': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_22': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_23': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_24': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_25': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_26': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_27': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_28': {'something': {'key': 'value', 'key2': 'value2'}}, 'key_29': {'something': {'key': 'value', 'key2': 'value2'}}}
